addons/binding/org.openhab.binding.innogysmarthome/src/main/java/org/openhab/binding/innogysmarthome/discovery/InnogyDeviceDiscoveryService.java

```diff
@@ -0,0 +1,198 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.innogysmarthome.discovery;
+
+import static org.openhab.binding.innogysmarthome.InnogyBindingConstants.*;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BridgeHandler;
+import org.openhab.binding.innogysmarthome.handler.InnogyBridgeHandler;
+import org.openhab.binding.innogysmarthome.handler.InnogyDeviceHandler;
+import org.openhab.binding.innogysmarthome.internal.listener.DeviceStatusListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import in.ollie.innogysmarthome.entity.device.Device;
+import in.ollie.innogysmarthome.entity.event.Event;
+
+/**
+ * The {@link InnogyDeviceDiscoveryService} is responsible for discovering new devices.
+ *
+ * @author Oliver Kuhl - Initial contribution
+ */
+public class InnogyDeviceDiscoveryService extends AbstractDiscoveryService implements DeviceStatusListener {
```

Comment:
You are right! I removed the interface here and changed the rest, see following comments below.